src/mono/sample/wasm/browser-nextjs/components/deepThought.js

```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect } from 'react'
+import createDotnetRuntime from '@microsoft/dotnet-runtime'
+
+let dotnetRuntime = undefined;
```

Comment:
React has a way to declare state that is shared between components. Loaded runtime is definitely something that fits in. I'm not sure if we should to keep it simple here or do it the right way.

Reply:
I improved the code, but I would like to keep it vanillaJs.